src/main.ts

```diff
@@ -58,113 +65,102 @@ async function getChangeSet(client: AxiosInstance, workspaceId: string) {
 
 type ChangeSet = Awaited<ReturnType<typeof getChangeSet>>
 
-async function setComponentProperties(
-  client: AxiosInstance,
-  { changeSetWebUrl, changeSetUrl }: ChangeSet
-) {
-  core.startGroup('Setting component properties ...')
-  // Get workspaceId from input or from whoami if there is no input
-  const componentId = core.getInput('componentId')
-  const domain = YAML.parse(core.getInput('domain'))
-  await client.put(`${changeSetUrl}/components/${componentId}/properties`, {
-    domain
-  })
-  core.setOutput(
-    'componentWebUrl',
-    `${changeSetWebUrl}?s=c_${componentId}&t=attributes`
-  )
-  core.endGroup()
+function parseInput(attributes: string) {
+  try {
+    return JSON.parse(attributes)
+  } catch {
+    return YAML.parse(attributes)
```

Comment:
... talk to me :)

Reply:
FYI, YAML is a strict superset of JSON, so you should be able to use YAML.parse() either way.